tests/BuildScriptGenerator.Tests/DotnetCore/DotNetCorePlatformTest.cs

```diff
@@ -19,82 +19,41 @@ public class DotNetCorePlatformTest
         public void Detect_ThrowsUnsupportedException_ForUnknownNetCoreAppVersion()
         {
             // Arrange
-            var projectFile = "test.csproj";
-            var projectFileContent = SampleProjectFileContents.ProjectFileWithTargetFrameworkPlaceHolder.Replace(
-                "#TargetFramework#",
-                "netcoreapp0.0");
-            var sourceRepo = new Mock<ISourceRepo>();
-            sourceRepo
-                .Setup(repo => repo.EnumerateFiles(It.IsAny<string>(), It.IsAny<bool>()))
-                .Returns(new[] { projectFile });
-            sourceRepo
-                .Setup(repo => repo.ReadFile(It.IsAny<string>()))
-                .Returns(projectFileContent);
-            var context = CreateContext(sourceRepo.Object);
-            var platform = CreatePlatform(projectFile: projectFile, defaultVersion: "2.2.7");
-
-            var exception = Assert.Throws<UnsupportedVersionException>(
-                () => platform.Detect(context));
-            Assert.Equal(
-                $"Platform 'dotnet' version '0.0' is unsupported. " +
-                "Supported versions: 2.2.7",
-                exception.Message);
-        }
+            var detector = CreateDetector(detectedVersion: "0.0");
+            var context = CreateContext();
+            var platform = CreatePlatform(
+                detector,
+                defaultVersion: "2.1.1",
+                supportedVersions: new Dictionary<string, string> { { "2.1.1", "3.1.3" } });
 
-        [Fact]
-        public void Detect_ThrowsUnsupportedException_WhenNoVersionFoundReturnsMaximumSatisfyingVersion()
```

Comment:
We still want to keep this one imo

Reply:
I removed this test case because it is already covered by the test above this one.